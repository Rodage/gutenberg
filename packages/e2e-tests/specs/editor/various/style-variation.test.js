/**
 * WordPress dependencies
 */
import {
	createNewPost,
	clickBlockToolbarButton,
	insertBlock,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';

describe( 'adding blocks', () => {
	beforeAll( async () => {
		await createNewPost();
	} );

	it( 'Should switch to the large style of the quote block', async () => {
		// Inserting a quote block
		await insertBlock( 'Quote' );
		await page.keyboard.press( 'ArrowDown' );
		await page.keyboard.type( 'Quote content' );

		// After adding content, the selected block
		// would be the paragraph, hence we have
		// to select the parent and then click for open
		// style variations.
		await clickBlockToolbarButton( 'Select Quote' );
		await clickBlockToolbarButton( 'Quote' );

		const largeStyleButton = await page.waitForXPath(
			'//*[@role="menuitem"][contains(., "Large")]'
		);
		await largeStyleButton.click();

		// Check the content
		const content = await getEditedPostContent();
		expect( content ).toMatchSnapshot();
	} );
} );
