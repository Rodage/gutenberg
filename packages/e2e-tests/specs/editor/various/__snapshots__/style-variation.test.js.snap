// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adding blocks Should switch to the large style of the quote block 1`] = `
"<!-- wp:quote {\\"className\\":\\"is-style-large\\"} -->
<blockquote class=\\"wp-block-quote is-style-large\\"><!-- wp:paragraph -->
<p>Quote content</p>
<!-- /wp:paragraph --></blockquote>
<!-- /wp:quote -->"
`;
